analyzers/src/SonarAnalyzer.CSharp/Rules/MessageTemplates/MessageTemplateAnalyzer.cs

```diff
@@ -51,7 +51,7 @@ public sealed class MessageTemplateAnalyzer : SonarDiagnosticAnalyzer
         },
         SyntaxKind.InvocationExpression);
 
-    private static ArgumentSyntax TemplateArgument(InvocationExpressionSyntax invocation, SemanticModel model) =>
+    public static ArgumentSyntax TemplateArgument(InvocationExpressionSyntax invocation, SemanticModel model) =>
```

Comment:
I need this call in the new `ShouldBeCorrect` rule, but I am not sure where to put it.
It refers to C# specific types, e..g `Invocation`, so I can't put it in `Common`.
Making it an extension method is also not great, since it will be exposed to everyone who uses `Invocation`.

I could move `MessageTemplates.cs` from `Common` to `CSharp` project and move this functionality there.
Realistically speaking, I do not see us implementing these rules in VB.NET.

And even if we do, moving it back shouldn't be that annoying.